### ISSUE TITLE
Swagger output with inheritance

### ISSUE THREAD

**Issue by patvong** [stoicflame/enunciate#374]
I have this inheritance chain:
```
public class Invoice extends InvoiceList {
   // ...
}

public class InvoiceList extends ResourceSupport {
   // ...
}

public class ResourceSupport {
   private final List<Link> links;
   // ...
}
```
In the generated `swagger.json`, I get:
```
"Invoice": {
      "type": "object",
      "title": "Invoice",
      "allOf": [
        {
          "$ref": "#/definitions/ResourceSupport"
        },
        {
          "$ref": "#/definitions/InvoiceListing"
        },
        {
          "properties": {
            }
          }
        }
      ],
      "description": ""
    },
```
Since `InvoiceListing` is defined like this:
```
"InvoiceListing": {
      "type": "object",
      "title": "InvoiceListing",
      "allOf": [
        {
          "$ref": "#/definitions/ResourceSupport"
        },
        {
          "properties": {
           }
        }
      ],
      "description": ""
}
```
Shouldn't `Invoice` definition have only one ref?
```
{
          "$ref": "#/definitions/InvoiceListing"
}
```

**Comment by patvong** [comment]
Is there any workaround for this at the moment?

**Comment by stoicflame** [comment]
Fixed at f795808.

**Comment by stoicflame** [comment]
[Enunciate 2.4 has been released](https://github.com/stoicflame/enunciate/releases/tag/v2.4.0).
